File: cpp/reverse-string/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string
{
    std::string reverse_string(std::string line)
    {
        char *reversed = new char[line.length() + 1];
        int c = 0;
        for (int i = line.length() - 1; i >= 0; i--)
        {
            reversed[c++] = line[i];
        }

        reversed[c] = '\0';
        std::string cc = reversed;
        return cc;
    }
} // namespace reverse_string

File: java/reverse-string/src/main/java/ReverseString.java
class ReverseString {

    String reverse(String inputString) {
        // StringBuilder 객체를 생성하고 입력 문자열을 전달
        StringBuilder sb = new StringBuilder(inputString);
        // reverse 메소드를 호출하여 문자열을 뒤집음
        sb.reverse();
        // 뒤집힌 문자열을 String으로 변환하여 반환
        return sb.toString();
    }

}
